docs: Add multiuser documentation and reposition Workflows (#9099)
invoke-ai/InvokeAI: docs/src/content/docs/features/Multi-User Mode/user-guide.mdx

Reviewer: Not sure we should leave the spec here, everything this covers is already documented. This can probably be moved to a readme within the invoke feature's directory.

Author: I removed the spec entirely. It was a planning document for copilot that I used at the beginning of the design process and was seriously out of date.

---
title: Multi-User Guide
description: How to use InvokeAI in multi-user mode as an end user.
sidebar:
  order: 3
---
import { Steps } from '@astrojs/starlight/components'

## Overview

Multi-User mode is a recent feature (introduced in version 6.12), which allows multiple individuals to share a single InvokeAI server while keeping their work separate and organized. Each user has their own username and login password, images, assets, image boards, customization settings and workflows.

Two types of users are recognized:

- A user with **Administrator** status can add, remove and modify other users, and can install models. They also have the ability to view the full session queue and pause or kill other users' jobs.

> - **Non-administrator** users can modify their own profile but not others. They also do not have the ability to install or configure models, but must ask an Administrator to do this task.

Reviewer: Non-admins can also view full queue but with other user details redacted

Author: Clarified.

---

## Getting Started

To activate Multi-User mode, open the `INVOKEAI_ROOT/invokeai.yaml` configuration file in a text editor. Add this line anywhere in the file:

```yaml
multiuser: true
```

You may also wish to make InvokeAI available to other machines on your local LAN. Add an additional line to `invokeai.yaml`:

```yaml
host: 0.0.0.0
```

Restart the server. It will now be in multi-user mode. If you enabled the `host` option, other users on your home or office LAN will be able to reach it by browsing to the IP address of the machine the backend is running on (`http://host-ip-address:9090`).

:::tip[Do not expose InvokeAI to the internet]
It is not recommended to expose the InvokeAI host to the internet due to security concerns.
:::

### Initial Setup (First Time in Multi-User Mode)

If you're the first person to access a fresh InvokeAI installation in multi-user mode, you'll see the **Administrator Setup** dialog:

![Administrator Setup Screen](./assets/admin-setup.png)

Now:

<Steps>
1. Enter your email address (this will be your login name)
2. Create a display name (this will be the name other users see)
3. Choose a strong password that meets the requirements:

>     - At least 8 characters long
>     - Contains uppercase letters
>     - Contains lowercase letters
>     - Contains numbers

Reviewer: ```suggestion
    With `strict_password_checking_enabled: true`:

      - At least 8 characters long
      - Contains uppercase letters
      - Contains lowercase letters
      - Contains numbers

    With `strict_password_checking_enabled` disabled, weaker passwords may still be accepted, but strong passwords are still strongly recommended.
```

Author: fixed

![Login Screen](./assets/user-login-1.png)

As an unprivileged user you can do pretty much anything that's allowed under single-user mode — generating images, using LoRAs, creating and running workflows, creating image boards — but you are restricted against installing new models, changing low-level server settings, or interfering with other users. More information on user roles is given below.

### Changing your Profile

To change your display name or profile, click on the person silhouette icon at the bottom left of the screen and choose "My Profile". This will take you to a screen that lets you change these values. At this time you can change your display name but not your login ID (ordinarily your contact email address).

---

## Understanding User Roles

In single-user mode, you have access to all features without restrictions. In multi-user mode, InvokeAI has two user roles:

### Regular User

As a regular user, you can:

- Create and manage your own image boards
- Generate images using all AI tools (Linear, Canvas, Upscale, Workflows)
- Create, save, and load your own workflows

> - View your own generation queue

Reviewer: This directly conflicts with line 131

Author: Fixed.

- View and use workflows marked as shared by other users

You cannot:

- Add, delete, or modify models
- View or modify other users' private boards, images, or workflows
- Manage user accounts
- Access system configuration
- View or cancel other users' generation tasks

:::tip[The generation queue]
When two or more users are accessing InvokeAI at the same time, their image generation jobs will be placed on the session queue on a first-come, first-serve basis. This means that you will have to wait for other users' image rendering jobs to complete before yours will start.

When another user's job is running, you will see the image generation progress bar and a queue badge that reads `X/Y`, where "X" is the number of jobs you have queued and "Y" is the total number of jobs queued, including your own and others.

You can also pull up the Queue tab in order to see where your job is in relationship to other queued tasks.
:::

### Administrator

Administrators have all regular user capabilities, plus:

- Full model management (add, delete, configure models)
- Create and manage user accounts
- View and manage all users' generation queues

> - View and manage all users' boards, images, and workflows (including system-owned legacy content)

Reviewer: ```suggestion
- View and manage all users' boards, images, and workflows
```

Author: Fixed.

Multiple users can be granted Administrator status.

---

## Getting Started

To activate Multi-User mode, open the `INVOKEAI_ROOT/invokeai.yaml` configuration file in a text editor. Add this line anywhere in the file:

```yaml
multiuser: true
```

You may also wish to make InvokeAI available to other machines on your local LAN. Add an additional line to `invokeai.yaml`:

```yaml
host: 0.0.0.0
```

Restart the server. It will now be in multi-user mode. If you enabled the `host` option, other users on your home or office LAN will be able to reach it by browsing to the IP address of the machine the backend is running on (`http://host-ip-address:9090`).

:::tip[Do not expose InvokeAI to the internet]
It is not recommended to expose the InvokeAI host to the internet due to security concerns.
:::

### Initial Setup (First Time in Multi-User Mode)

If you're the first person to access a fresh InvokeAI installation in multi-user mode, you'll see the **Administrator Setup** dialog:

![Administrator Setup Screen](./assets/admin-setup.png)

Now:

<Steps>
1. Enter your email address (this will be your login name)
2. Create a display name (this will be the name other users see)

> 3. Choose a strong password. The following criteria are required with `strict_password_checking_enabled: true`.

Reviewer: ```suggestion
3. Choose a strong password. The following criteria are required with `strict_password_checking: true`.
```
My bad on my last suggestion lol

    - At least 8 characters long
    - Contains uppercase letters
    - Contains lowercase letters
    - Contains numbers
4. Confirm your password
5. Click **Create Administrator Account**
</Steps>

> With `strict_password_checking_enabled` disabled, you'll be warned if you choose a

Reviewer: ```suggestion
With `strict_password_checking` disabled, you'll be warned if you choose a
```
My bad on my earlier suggestion

weak password, but not prevented from doing so.

You'll now be taken to a login screen and can enter the credentials you just created.

### Adding and Modifying Users

> If you are logged in as Administrator, you can add additional users. Click on the small "person silhouette" icon at the bottom left of the main Invoke screen and select "User Management:"

Reviewer: ```suggestion
If you are logged in as Administrator, you can add additional users. Click on the small "person silhouette" icon at the bottom left of the main Invoke screen and select "User Management"
```

- Full model management (add, delete, configure models)
- Create and manage user accounts
- View and manage all users' generation queues
- View and manage all users' boards, images, and workflows (including system-owned legacy content)
- Access system configuration
- Grant or revoke admin privileges

---

## Working with Your Content in Multi-User Mode

### Image Boards

In multi-user mode, each user can create an unlimited number of boards and organize their images and assets as they see fit. Boards have three visibility levels:

- **Private** (default): Only you (and administrators) can see and modify the board.
- **Shared**: All users can view the board and its contents, but only you (and administrators) can modify it (rename, archive, delete, or add/remove images).
- **Public**: All users can view the board. Only you (and administrators) can modify the board's structure (rename, archive, delete).

To change a board's visibility, right-click on the board and select the desired visibility option.

Administrators can see and manage all users' image boards and their contents regardless of visibility settings.

### Going From Multi-User to Single-User Mode

> If an InvokeAI instance was in multiuser mode and then restarted in single user mode (by setting `multiuser: false` in the configuration file), all users' boards will be consolidated in one place. Any images that were in "Uncategorized" will be merged together into a single Uncategorized board. If, at a later date, the server is restarted in multi-user mode, the boards and images will be separated and restored to their owners.

Reviewer: ```suggestion
If an InvokeAI instance was in multiuser mode and then restarted in single user mode (by setting `multiuser: false` in the configuration file), all users' boards will be consolidated in one place. Any images that were in "Uncategorized" will be merged together into a single Uncategorized board. If, at a later date, the server is restarted in multi-user mode, the boards and images will be assigned to the internal 'system' user. Admins can access this legacy content, and will not be restored to original owners.
```

Multiple users can be granted Administrator status.

---

## Getting Started

To activate Multi-User mode, open the `INVOKEAI_ROOT/invokeai.yaml` configuration file in a text editor. Add this line anywhere in the file:

```yaml
multiuser: true
```

You may also wish to make InvokeAI available to other machines on your local LAN. Add an additional line to `invokeai.yaml`:

```yaml
host: 0.0.0.0
```

Restart the server. It will now be in multi-user mode. If you enabled the `host` option, other users on your home or office LAN will be able to reach it by browsing to the IP address of the machine the backend is running on (`http://host-ip-address:9090`).

:::tip[Do not expose InvokeAI to the internet]
It is not recommended to expose the InvokeAI host to the internet due to security concerns.
:::

### Initial Setup (First Time in Multi-User Mode)

If you're the first person to access a fresh InvokeAI installation in multi-user mode, you'll see the **Administrator Setup** dialog:

![Administrator Setup Screen](./assets/admin-setup.png)

Now:

> 

Reviewer: Let's use the `<Steps>` components for instructional ordered lists like these.

https://starlight.astro.build/components/steps/

Author: I've used <Steps> everywhere where there was a numbered list. I like the formatting you get!

### Adding and Modifying Users

If you are logged in as Administrator, you can add additional users. Click on the small "person silhouette" icon at the bottom left of the main Invoke screen and select "User Management:"

![Administrator Menu](./assets/admin-add-user-1.png)

This will take you to the User Management screen...

![User Management screen](./assets/admin-add-user-2.png)

...where you can click "Create User" to add a new user.

![Add User Screen](./assets/admin-add-user-3.png)

The User Management screen also allows you to:

1. Temporarily change a user's status to Inactive, preventing them from logging in to Invoke.
2. Edit a user (by clicking on the pencil icon) to change the user's display name or password.
3. Permanently delete a user.
4. Grant a user Administrator privileges.

### Command-line User Management Scripts

Administrators can also use a series of command-line scripts to add, modify, or delete users. If you use the launcher, click the ">" icon to enter the command-line interface. Otherwise, if you are a native command-line user, activate the InvokeAI environment from your terminal.

The commands are named:

> 

Reviewer: A table would be good here

Author: Done.

When you enable multi-user mode for the first time on an existing InvokeAI installation, all workflows that were created before multi-user mode was activated will appear in the **shared workflows** section. These preexisting workflows are owned by the internal "system" account and are visible to all users. Administrators can edit or delete these shared legacy workflows. Regular users can view and use them but cannot modify them.
:::

### The Generation Queue

The queue shows your pending and running generation tasks.

**Queue Features:**

- View your current and completed generations
- Cancel pending tasks
- Re-run previous generations
- Monitor progress in real-time

**Queue Isolation:**

- You will see your own queue items, as well as the items generated by other users, but the generation parameters (e.g. prompts) for other users' jobs are hidden for privacy reasons.
- Administrators can view all queues for troubleshooting.
- Your generations won't interfere with other users' tasks.

---

## Customizing Your Experience

### Personal Preferences

Your UI preferences are saved to your account and are restored when you log in:

> - **Theme**: Choose between light and dark modes

Reviewer: I don't think we have a configurable light/dark theme

Author: Fixed. That one looks like an LLM hallucination. Most of the other inconsistencies are due to out-of-sync modifications to the specifications.
